mechanisms/generate_catalogue

```diff
@@ -120,7 +120,7 @@ const mechanism_catalogue& global_${catalogue}_catalogue() {
 
 #ifdef STANDALONE
 extern "C" {
-    const void* get_catalogue() {
+    __attribute__ ((visibility ("default"))) const void* get_catalogue() {
```

Comment:
Since C++17 we are allowed to use `[[gnu::visibility("default")]]` as part of the standard. The `__attribute__` is technically an extension by GCC.

Reply:
But that would only work for GCC. According to this: https://gcc.gnu.org/wiki/Visibility, we should add this entire block:
```
#if defined _WIN32 || defined __CYGWIN__
  #ifdef BUILDING_DLL
    #ifdef __GNUC__
      #define DLL_PUBLIC __attribute__ ((dllexport))
    #else
      #define DLL_PUBLIC __declspec(dllexport) // Note: actually gcc seems to also supports this syntax.
    #endif
  #else
    #ifdef __GNUC__
      #define DLL_PUBLIC __attribute__ ((dllimport))
    #else
      #define DLL_PUBLIC __declspec(dllimport) // Note: actually gcc seems to also supports this syntax.
    #endif
  #endif
  #define DLL_LOCAL
#else
  #if __GNUC__ >= 4
    #define DLL_PUBLIC __attribute__ ((visibility ("default")))
    #define DLL_LOCAL  __attribute__ ((visibility ("hidden")))
  #else
    #define DLL_PUBLIC
    #define DLL_LOCAL
  #endif
#endif
```
I think we need that top block for WSL, but I'm not sure and I don't have a system to test it on.

Reply:
That page predates the C++17 attribute (last change 2016). We also can savely assume GCC > 4 (else C++14 would be tricky). And finally, we do not care about Windows right now. In endeffect it comes down to what you have in the PR.

Regarding your comment, this is working in clang and gcc. To quote my last comment `__attribute__` is a GCC extension in itself. However, Clang shares most of GCC's extensions. Reference: https://gcc.godbolt.org/z/Edev9PGnK